File: problems-adapt/2101-2200/2125_beams-across-empty-floors/solution.cpp
class Solution {
  public:
    int crossBeams(vector<string> &floors) {
        int beams = 0;
        int previous = 0;
        for (const string &row : floors) {
            int devices = count(row.begin(), row.end(), '1');
            if (devices > 0) {
                beams += previous * devices;
                previous = devices;
            }
        }
        return beams;
    }
};

File: problems-adapt/2101-2200/2125_beams-across-empty-floors/solution.py
from typing import List


class Solution:
    def crossBeams(self, floors: List[str]) -> int:
        beams = 0
        previous = 0
        for row in floors:
            devices = row.count("1")
            if devices > 0:
                beams += previous * devices
                previous = devices
        return beams
